src/core/components/collapsible/collapsible.tsx

```diff
@@ -22,27 +22,56 @@ export const Collapsible: React.FC<ICollapsibleProps> = (props) => {
     const [isOpenState, setIsOpenState] = useState(defaultOpen);
     const [isOverflowing, setIsOverflowing] = useState(false);
     const [maxHeight, setMaxHeight] = useState(0);
+    const [overlayMinHeight, setOverlayMinHeight] = useState<number>(112);
 
     const defaultLineHeight = 24;
 
     const isOpen = isOpenProp ?? isOpenState;
     const contentRef = useRef<HTMLDivElement>(null);
 
     const calculateCollapsedHeight = useCallback(() => {
-        if (collapsedPixels != null) {
-            return collapsedPixels;
-        }
+        let collapsedHeightBase: number;
 
-        if (collapsedLines && contentRef.current) {
+        if (collapsedPixels != null) {
+            collapsedHeightBase = collapsedPixels;
+        } else if (collapsedLines && contentRef.current && typeof window !== 'undefined') {
             const lineHeight = parseFloat(window.getComputedStyle(contentRef.current).lineHeight);
-
             if (!Number.isNaN(lineHeight)) {
-                return lineHeight * collapsedLines;
+                collapsedHeightBase = lineHeight * collapsedLines;
+            } else {
+                collapsedHeightBase = collapsedLines * defaultLineHeight;
             }
+        } else {
+            collapsedHeightBase = collapsedLines * defaultLineHeight;
+        }
+
+        if (showOverlay && !isOpen) {
+            // Clamp to the current overlay min height
+            return Math.max(collapsedHeightBase, overlayMinHeight);
+        }
+
+        return collapsedHeightBase;
+    }, [collapsedPixels, collapsedLines, defaultLineHeight, showOverlay, isOpen, overlayMinHeight]);
+
+    useEffect(() => {
+        if (!showOverlay) {
+            return;
         }
 
-        return collapsedLines * defaultLineHeight;
-    }, [collapsedPixels, collapsedLines, defaultLineHeight]);
+        const mdBreakpointPx = 768; // Tailwind default 'md'
+        const updateOverlayMinHeight = () => {
+            if (typeof window === 'undefined') {
+                return;
+            }
+            // Tailwind sizes: h-28 (base) => 7rem (112px), md:h-32 => 8rem (128px)
+            const next = window.innerWidth >= mdBreakpointPx ? 128 : 112;
+            setOverlayMinHeight(next);
+        };
```

Comment:
Wondering if we can make the const a default as a more generic prop here? This component is always problematic since it's pretty specific and has a lot of considerations, but as a component library we try to leave it as unopinionated with hard coded values as possible.